### DISCUSSION TITLE
useMutation with Next.js API route error handling

### DISCUSSION THREAD

**Discussion by zi-gae** [TanStack/query#1354] [General]
There is no way to get the different error messages returned by the endpoint?

```ts
try {
  const { status, data } = await axios(options)
  return res.status(status).json(data)
} catch (error) {
  next(error.response)
}
```
The `error.response` value I want to receive on the endpoint

Values on the Response tab of Chrome develop tools
```html
<!DOCTYPE html>
<html lang="en">
<head>
<meta charset="utf-8">
<title>Error</title>
</head>
<body>
<pre>[object Object]</pre>
</body>
</html>
```

**Comment by TkDodo** [comment]
> The error.response value I want to receive on the endpoint

I don’t know next js well, but since the error comes from axios, i think it’s just an axiosError. The normal error message should be at `error.message`, and depending on what your actual error returns (a json body maybe?), you can find that inside the response, e.g. `error.response.data`
